plugins/toolkit-activation/metric/new_histogram_intercepter.go

```diff
@@ -0,0 +1,45 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package metric
+
+import (
+	"github.com/apache/skywalking-go/plugins/core/metrics"
+	"github.com/apache/skywalking-go/plugins/core/operator"
+	"github.com/apache/skywalking-go/toolkit/metric"
+)
+
+type NewHistogramInterceptor struct{}
+
+func (h *NewHistogramInterceptor) BeforeInvoke(_ operator.Invocation) error {
+	return nil
```

Comment:
Please using the defined result when interceptor all metrics constructor. It's better to control all object in the same side, if we change the toolkit API or agent, it could have unknown dependency error.

Reply:
I don't quite understand this, are you saying that we shouldn't use anonymous function parameters or do you mean something else?

Reply:
I mean is it possible we can construct the metrics by the plugin?

Reply:
This approach may be possible, but we need to consider the following issues:
1. Parameter parsing is necessary in the interceptor. Even if we build it in the plugin, we still need to parse the parameters
2. Using (1) to build a metric instance, if it is implemented in the plugin, will it introduce unnecessary redundant code?
3. In fact, we don’t do anything complicated in the interceptor, as I said above: parse parameters -> build objects. When the dependent `NewXXX` changes, compatibility must be considered
So I think we may not need to change this

Reply:
Why it's related to the parameter? I just don't want to use the existing instance created from the toolkit API. 
The basic process is: to build and define the result in the `BeforeInvoke`, then the toolkit implementation will not be invoked. The plugin controls all (toolkit)metrics instances, if we want to extend the metrics it's easy to adapt.